docroot/sites/classrooms.uiowa.edu/modules/classrooms_core/classrooms_core.module

```diff
@@ -44,4 +68,32 @@ function classrooms_core_entity_presave(EntityInterface $entity): void {
     $title = "{$building->label()} $room_id";
     $entity->set('title', $title);
   }
+
+  // Grab MAUI room data.
+  $maui_api = \Drupal::service('uiowa_maui.api');
+  $data = $maui_api->getRoomData($building_id, $room_id);
+  if ($data) {
+    // Set the Maximum Occupancy field
+    // if data exists/integer and not overridden.
+    if ($entity->hasField('field_room_max_occupancy') && isset($data[0]->maxOccupancy)) {
+      if (filter_var($data[0]->maxOccupancy, FILTER_VALIDATE_INT) !== FALSE) {
+        $entity->set('field_room_max_occupancy', $data[0]->maxOccupancy);
+      }
+    }
+    if ($entity->hasField('field_room_name') && isset($data[0]->roomName)) {
+      if (strlen($data[0]->roomName) > 1) {
+        $entity->set('field_room_name', $data[0]->roomName);
+      }
+    }
+
+    // Select list field not working yet.
+    //    if ($entity->hasField('field_room_instruction_category') && isset($data[0]->roomCategory)) {
+    //      $field_definition = $entity->getFieldDefinition('field_room_instruction_category')->getFieldStorageDefinition();
+    //      $field_allowed_options = options_allowed_values($field_definition, $entity);
+    //      if (in_array($data[0]->roomCategory, $field_allowed_options)) {
```

Comment:
```suggestion
    //      if (array_key_exists($data[0]->roomCategory, $field_allowed_options)) {
```

couldn't leave without understanding why this wouldn't work. we want to compare the key, not the array value.